src/backend/distributed/metadata/node_metadata.c

```diff
@@ -273,6 +273,26 @@ citus_add_node(PG_FUNCTION_ARGS)
 	 */
 	if (!nodeAlreadyExists)
 	{
+		WorkerNode *workerNode = FindWorkerNodeAnyCluster(nodeNameString, nodePort);
+
+		/*
+		 * If the worker is not marked as a coordinator, check that
+		 * the node is not trying to add itself
+		 */
+		if (workerNode != NULL &&
+			workerNode->groupId != 0 &&
+			IsWorkerTheCurrentNode(workerNode))
+		{
+			ereport(ERROR, (errmsg("Node cannot add itself as a worker."),
```

Comment:
maybe we give the hint based on the state, giving both citus_add_node and citus_set_coordinator_host as suggestion is confusing

```C

	WorkerNode *coordinatorNode = PrimaryNodeForGroup(COORDINATOR_GROUP_ID,
													  &isCoordinatorInMetadata);
     if (isCoordinatorInMetadata) hint_with_citus_set_coordinator_host
     else hint_with_ citus_add_node

Reply:
(and, also add test triggering both)

Reply:
Afaik If the coordinator is in the metadata, an error will be thrown because coordinator permissions are needed to run citus_add_node and this code I added will not execute.

Reply:
> an error will be thrown because coordinator permissions are needed

can you please clarify this a bit more? I can do it without an error:
```SQL
# SELECT nodename,nodeport, groupid FROM pg_dist_node;
┌───────────┬──────────┬─────────┐
│ nodename  │ nodeport │ groupid │
├───────────┼──────────┼─────────┤
│ localhost │     5432 │       0 │
└───────────┴──────────┴─────────┘
(1 row)

SELECT citus_add_node('localhost', 5432);
┌────────────────┐
│ citus_add_node │
├────────────────┤
│             10 │
└────────────────┘
(1 row)
```

I think we don't get here because we are operating on `if (!nodeAlreadyExists))`, where the node already exists.

What we can do it:
1) remove `citus_set_coordinator_host` from this error message
2) on the else of `if (!nodeAlreadyExists)`, if `isCoordinatorInMetadata`, we give a NOTICE to use `citus_set_coordinator_host`.

Reply:
> What we can do it remove citus_set_coordinator_host from this error message


As Marco noted in the other comment in this PR, we decided to only go with `citus_set_coordinator_host`